com.unity.visualeffectgraph/Editor/Controls/VFXMinMaxSliderField.cs

```diff
@@ -0,0 +1,199 @@
+using System;
+using UnityEngine;
+using UnityEngine.UIElements;
+using UnityEditor.UIElements;
+using System.Collections.Generic;
+using System.Linq;
+
+using Action = System.Action;
+
+namespace UnityEditor.VFX.UI
+{
+    class VFXMinMaxSliderField : VisualElement, INotifyValueChanged<Vector2>
+    {
+        protected MinMaxSlider m_Slider;
+        VisualElement m_IndeterminateLabel;
+
+        class StartFinishSliderManipulator : Manipulator
+        {
+            VFXMinMaxSliderField m_Slider;
+            bool m_InDrag;
+            protected override void RegisterCallbacksOnTarget()
+            {
+                m_Slider = target.GetFirstOfType<VFXMinMaxSliderField>();
+
+                target.RegisterCallback<MouseDownEvent>(OnMouseDown, TrickleDown.TrickleDown);
+                target.RegisterCallback<MouseUpEvent>(OnMouseUp, TrickleDown.TrickleDown);
+            }
+
+            protected override void UnregisterCallbacksFromTarget()
+            {
+                target.UnregisterCallback<MouseDownEvent>(OnMouseDown, TrickleDown.TrickleDown);
+                target.UnregisterCallback<MouseUpEvent>(OnMouseUp, TrickleDown.TrickleDown);
+            }
+
+            void OnMouseDown(MouseDownEvent e)
+            {
+                target.RegisterCallback<MouseMoveEvent>(OnMouseMove, TrickleDown.TrickleDown);
+            }
+
+            void OnMouseMove(MouseMoveEvent e)
+            {
+                if (!m_InDrag)
+                {
+                    m_InDrag = true;
+                    m_Slider.ValueDragStarted();
+                    target.UnregisterCallback<MouseMoveEvent>(OnMouseMove, TrickleDown.TrickleDown); //we care only about the first drag event
+                }
+            }
+
+            void OnMouseUp(MouseUpEvent e)
+            {
+                if (m_InDrag)
+                    m_Slider.ValueDragFinished();
+                else
+                    target.UnregisterCallback<MouseMoveEvent>(OnMouseMove, TrickleDown.TrickleDown);
+                m_InDrag = false;
+            }
+        }
+
+        protected void ValueDragFinished()
+        {
+            if (onValueDragFinished != null)
+                onValueDragFinished();
+        }
+
+        protected void ValueDragStarted()
+        {
+            if (onValueDragStarted != null)
+                onValueDragStarted();
+        }
+
+        public Action onValueDragFinished;
+        public Action onValueDragStarted;
+
+        public VFXMinMaxSliderField()
+        {
+            AddToClassList("sliderMinMaxField");
+            m_Slider = new MinMaxSlider(1, 10, 0, 100);
+
+            m_Slider.RegisterValueChangedCallback(evt => ValueChanged(evt.newValue));
+            m_IndeterminateLabel = new Label()
+            {
+                name = "indeterminate",
```

Comment:
I wonder if the string that will be displayed in the UI should support "localization"?
I know `EditorGUIUtility.TrTextContent` is used for IMGUI but I don't know if it's compatible with VisualElement API

Reply:
I actually don't know much about localization, but I see no sign of it in the other files...